packages/mrm-task-eslint/index.js

```diff
@@ -60,19 +60,32 @@ function task(config) {
 
 	// TypeScript
 	if (pkg.get('devDependencies.typescript')) {
-		const parser = 'typescript-eslint-parser';
-		packages.push(parser);
+		const parser = '@typescript-eslint/parser';
+		const plugin = '@typescript-eslint/eslint-plugin';
+		packages.push(parser, plugin);
 		eslintrc.merge({
 			parser,
-			rules: eslintRules || {
-				// Disable rules not supported by TypeScript parser
-				// https://github.com/eslint/typescript-eslint-parser#known-issues
-				'no-undef': 0,
-				'no-unused-vars': 0,
-				'no-useless-constructor': 0,
+			plugins: [plugin],
+			parserOptions: {
+				// If using React, turn on JSX support in the TypeScript parser.
+				...(pkg.get('dependencies.react') && { ecmaFeatures: { jsx: true } }),
+				project: './tsconfig.json',
 			},
+			rules: eslintRules || {},
 		});
 		exts = ' --ext .ts,.tsx';
+
+		if (pkg.get('devDependencies.prettier')) {
```

Comment:
Why do we need this?

Reply:
It's debatable whether this should go in this task or the Prettier task. The reason I added it is because if you _are_ using Prettier then it is pretty common for several of the ESLint rules to conflict with Prettier's rules to the point it is actually impossible to satisfy both tools simultaneously. The solution to this is to extend from the `prettier` ESLint preset, but since there's a new TypeScript parser you also have to extend from `prettier/@typescript-eslint` to support it.

This is common and rote enough that I felt it was a nice affordance to set it up automatically if we can detect it's needed. There is some question of a race condition here, though, given the user could run the `prettier` and `eslint` tasks in either order.

Reply:
Yeah, that make sense. Though I think it shouldn't be the default behavior, since it's not needed for all ESLint presets.

Reply:
That's true, but there is some overlap between the rules in `eslint:recommended`, which we use as the default rule set for extension, and the rules that are turned off for compatibility in `eslint-config-prettier`. I suppose it is _possible_ that the user might provide their own via the `eslintPreset` config property that just happens to make `eslint-config-prettier` superfluous but in our present default behavior it is a relevant package.

I will concede and remove it if you insist but, at least from my experience, I've found I always end up adding it manually pretty soon afterward using this task.

**References**
- [`eslint:recommended` definition](https://github.com/eslint/eslint/blob/master/conf/eslint-recommended.js)
- [`eslint-config-prettier` definition](https://github.com/prettier/eslint-config-prettier/blob/master/index.js)

Reply:
I don't have a strong opinion to be honest, let's keep it for now ;-)